### ISSUE TITLE
Loading entities without OrderBy and Skip/Take

### ISSUE THREAD

**Issue by joacar** [dotnet/efcore#13155]
Loading collection using `DbContext.Entry().Load()` where query cotains Skip/Take **and** OrderBy.

### Steps to reproduce
[EFCoreLoadTakeOrderBy.zip](https://github.com/aspnet/EntityFrameworkCore/files/2333708/EFCoreLoadTakeOrderBy.zip)

### Further technical details
EF Core version: 2.1.2
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: 
IDE: Visual Studio 2017 15.5.x
Windows 10

**Comment by ajcvickers** [comment]
@joacar When I run this I get an exception that says I'm using Take without ordering. Query is:
```C#
ctx.Entry(instagram)
                   .Collection(_ => _.Posts)
                   .Query()
                   .Take(1)
                   .OrderBy(_ => _.CreatedDateTime)
                   .Load();
```
Since Take is before OrderBy the exception makes sense to me. Ordering and then Taking doesn't throw the exception:
```C#
ctx.Entry(instagram)
                   .Collection(_ => _.Posts)
                   .Query()
                   .OrderBy(_ => _.CreatedDateTime)
                   .Take(1)
                   .Load();
```
Again, as expected.

So what is the bug/question here?

**Comment by joacar** [comment]
Hi @ajcvickers 

My apologizes for beeing a bit trigger-happy and not providing more details. I was pretty sure that I tested the other way around last night, but apparently not! I guess I'm thinking SQL-ordering `select **top** .. **order by**` and not LINQ-ordering.

Just seeing the generated SQL made me think it was incorrect since it doesn't make sense to put the `order by` on the outer statement when it belongs to the inner, which should be unambiguous  for the above statement.

Unrelated question: Is there any way to configure EF as "TreatWarningsAsError"? I.e. it should throw an exception for every warning. Example code `ConfigureWarnings(config => config.Throw(ThreatWarningsAsErrors)`.

Closing this
